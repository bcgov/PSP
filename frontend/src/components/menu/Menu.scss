.Menu-root {
  position: relative;
  border: 0.1rem;
  display: flex;
  align-items: center;
  flex-direction: column;

  .Menu-button {
    font-size: 1.4rem;
  }

  .Menu-items {
    box-shadow: 0.2rem 0.2rem 0.5rem #888888;
    background-color: #fff;
    position: absolute;
    display: none !important;
    z-index: 700;
    flex-direction: column;
    &.open {
      display: flex !important;
    }

    &.left {
      left: 0;
    }

    &.top {
      bottom: 0;
    }

    .Menu-filter {
      margin: 0.5rem;
    }

    .Menu-options {
      box-shadow: 0.2rem 0.2rem 0.5rem #888888;
      z-index: 1000;
      &.scrollable {
        max-height: 22rem;
        overflow-y: scroll;
      }
    }

    .Menu-item {
      width: 100% !important;
      border-radius: 0rem !important;
      padding: 0.8rem 1.4rem;
      font-size: 1.2rem;
      text-align: left;
      &:hover {
        background-color: lighten(grey, 45%);
      }
      &:active {
        background-color: #ffc107 !important;
      }
    }
  }
}
