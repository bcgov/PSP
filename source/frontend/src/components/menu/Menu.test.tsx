import { fireEvent, render } from '@testing-library/react';

import { IMenuItemProps, Menu } from './Menu';

const mockFunction = jest.fn();

const mockOptions: IMenuItemProps[] = [
  {
    label: 'Option 1',
    onClick: mockFunction,
  },
  { label: 'Option 2' },
];
it('renders correctly', () => {
  const { asFragment } = render(<Menu options={mockOptions} />);
  expect(asFragment()).toMatchSnapshot();
});

it('calls appropriate function on click', () => {
  const { getByText } = render(<Menu options={mockOptions} />);
  fireEvent.click(getByText('Option 1'));
  expect(mockFunction).toHaveBeenCalledTimes(1);
});

it('renders label correctly', () => {
  const { getByText } = render(<Menu options={mockOptions} label="Test Label" />);
  expect(getByText('Test Label')).toBeInTheDocument();
});

it('renders all options correctly', () => {
  const { getByText } = render(<Menu options={mockOptions} />);
  expect(getByText('Option 1')).toBeInTheDocument();
  expect(getByText('Option 2')).toBeInTheDocument();
});
