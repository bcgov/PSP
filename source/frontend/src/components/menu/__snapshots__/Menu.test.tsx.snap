// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<DocumentFragment>
  <div
    class="Menu-root"
  >
    <div
      class="Menu-button"
    >
      <span>
        Menu
      </span>
    </div>
    <div
      class="Menu-items"
      style="width: 10.0rem;"
    >
      <div
        class="Menu-options scrollable list-group"
      >
        <div
          class="Menu-item list-group-item"
          title="menu-item-Option 1"
        >
          Option 1
        </div>
        <div
          class="Menu-item list-group-item"
          title="menu-item-Option 2"
        >
          Option 2
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
